### PR TITLE
[TRA-481] change vault order skewing to non-linear

### PR DESCRIPTION
**Pull request by tqin7**
### Changelist
implement vaults phase 2 skewing

### Test Plan
unit tests

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Summary by CodeRabbit

- **New Features**
  - Enhanced order calculation logic using refined formulas for better precision.
  - Adjusted order size determination to improve accuracy and performance.

- **Bug Fixes**
  - Corrected the computation of skew factors based on order side and leverage, ensuring better alignment with market conditions.

### CODE REVIEW

File: protocol/x/vault/keeper/orders.go [hunk 1]
```diff
@@ -269,51 +280,66 @@ func (k Keeper) GetVaultClobOrders(
...
+			subticks = lib.BigDivCeil(subticks, oracleSubticks.Denom())
 		} else {
-			subticks = new(big.Int).Quo(orderSubticksNum, oracleSubticks.Denom())
+			subticks = new(big.Int).Quo(subticks, oracleSubticks.Denom())
 		}
+		subticks = lib.BigMulPpm( // delayed division by 1 million (rounding up) as noted above.
+			subticks,
+			big.NewInt(1),
+			side == clobtypes.Order_SIDE_SELL,
+		)
```
Comment (BrendanChou) [inline protocol/x/vault/keeper/orders.go:341]: suggestion:
```
    divisor := lib.BigIntOneMillion()
    divisor.Mul(divisor, oracleSubticks.Denom())
		if side == clobtypes.Order_SIDE_SELL {
			subticks = lib.BigDivCeil(subticks, divisor)
		} else {
			subticks = subticks.Quo(subticks, divisor)
		}
```

File: protocol/x/vault/keeper/orders.go [hunk 2]
```diff
@@ -269,51 +280,66 @@ func (k Keeper) GetVaultClobOrders(
...
-		if side == clobtypes.Order_SIDE_SELL && spreadSkewPpm.Sign() < 0 {
-			spreadSkewPpm.SetUint64(0)
-		} else if side == clobtypes.Order_SIDE_BUY && spreadSkewPpm.Sign() > 0 {
-			spreadSkewPpm.SetUint64(0)
+		// Calculate skewed spread.
+		skewedSpreadPpmI := lib.BigIntOneMillion()
+		if side == clobtypes.Order_SIDE_SELL {
+			skewedSpreadPpmI.Add(skewedSpreadPpmI, skewPpmI)
+		} else {
+			skewedSpreadPpmI.Sub(skewPpmI, skewedSpreadPpmI)
```
Comment (BrendanChou) [inline protocol/x/vault/keeper/orders.go:320]: Is this meant to be:
```
skewedSpreadPpmI.Sub(skewedSpreadPpmI, skewPpmI)
```
?

If so, why doesn't a test catch this?

Comment (tqin7) [reply protocol/x/vault/keeper/orders.go:320]: intentionally negated this as 
```
a_i = oraclePrice * (1 + ask_spread_i)
b_i = oraclePrice * (1 - bid_spread_i)
```

so i can just do `1 + skewedSpreadPpmI` for both cases

**Review by sr33j — APPROVED** [review]
lgtm!

**Pull request merged**